css/amp.css

```diff
@@ -174,10 +174,21 @@ html.i-amphtml-singledoc.i-amphtml-ios-embed-sd > body {
   transition: opacity 0.3s cubic-bezier(0.0, 0.0, 0.2, 1) !important;
 }
 
-.i-amphtml-layout-nodisplay,
-[layout=nodisplay]:not(.i-amphtml-layout-nodisplay)
+/* layout=nodisplay are automatically hidden until JS initializes. */
+[layout=nodisplay]:not(.i-amphtml-element) {
+  display: none !important;
+}
+
+/* Initialized layout=nodisplay contain [hidden] */
+.i-amphtml-layout-nodisplay[hidden],
+[layout=nodisplay][hidden]:not(.i-amphtml-layout-nodisplay)
+{
+  /* Display is set/reset via the hidden attribute */
```

Comment:
What's the point of these selectors? Is this just a documentation thing?

Reply:
Yup. We have similar ones for [`.not-built`](https://github.com/ampproject/amphtml/blob/175102234f0d11d2236d1e2e897294ca621d49d0/css/amp.css#L363-L365), [`.i-amphtml-layout`](https://github.com/ampproject/amphtml/blob/175102234f0d11d2236d1e2e897294ca621d49d0/css/amp.css#L385-L387), and [`.i-amphtml-error`](https://github.com/ampproject/amphtml/blob/175102234f0d11d2236d1e2e897294ca621d49d0/css/amp.css#L389-L391).

I can remove if you'd prefer?

Reply:
I see. It's ok then. I assume optimizer simply strips these.